controller/adr.rst

```diff
@@ -0,0 +1,206 @@
+
+        $collection->add('hello', new Route('/hello', array(
+            '_controller' => HelloAction::class,
+        )));
+
+Creating a Responder
+--------------------
+
+As you can see in the __invoke call, this action require a ``HelloResponder`` class in order to build the response which is returned to the browser, first, update the services.yaml according to this need: 
+
+.. code-block:: yaml
+
+    parameters:
+        locale: 'en'
+
+    services:
+        _defaults:
+            autowire: true
+            autoconfigure: true
+            public: false
+
+        App\Actions\:
+            resource: '../src/Actions'
+            tags: 
+                - 'controller.service_arguments'
+                
+        App\Responders\:
+            resource: '../src/Responders'
+            
+Here, the container only need to know about the existence of the classes, nothing difficult to understand as the fact that our Responders are responsable of returning the actual Response to the browser, no need to add the 'controller.service_arguments' tags as the Responders need to be called using the __invoke method in order to receive data from the Action. 
+
+Now that the logic behind is clear, time to create the ``HelloResponder.php`` file: 
+                
+.. code-block:: php
+
+    <?php
+    
+    namespace App\Responders;
+
+    use Twig\Environment;
+    use Symfony\Component\HttpFoundation\Response;
+
+    final class HomeResponder
+    {
+        private $twig;
+        
+        public function __construct(Environment $twig)
+        {
+            $this->twig = $twig;
+        }
+
+        public function __invoke(array $data)
+        {
+            return new Response(
+                $this->twig->render('index.html.twig', $data)
+            );
+        }
+    }
+
+If the routing is clearly define, the browser should display the traditional "Hello World" using the ADR approach, congrats !
+
+Accessing the request
+---------------------
+
+In many case, your classes can ask for any data passed via a form or via an API call, 
+as you can imagine, as the logic evolve, your class is capable of accessing the request 
+from a simple method injection like this:
+
+.. code-block:: php
+
+    <?php
+
+        use Symfony\Component\HttpFoundation\Request;
+        // ...
+
+        public function __invoke(Environment $twig, Request $request): Response
+        {
+            $id = $request->get('id');
+            
+            return $twig->render('default/index.html.twig', array('id' => $id));
+        }
+    }
+    
+Final thought
+-------------
+
+Keep in mind that this approach can be completely different from what you're used to use, in order to
+keep your code clean and easy to maintain, we recommend to use this approach only if your code is
+decoupled from the internal framework logic (like with Clean Architecture approach) or if you start a new
+project and need to keep the logic linked to your business rules.
```

Comment:
I don't understand this last paragraph and the supposed benefits. I don't see how this approach or what you refer to the old one is cleaner or easier. I don't understand why this approach is more decoupled than the other one. I'm not saying this is good or bad, just that from a beginner's perspective, the benefits are not very clear.